mvc/models/model-binding.rst

```diff
@@ -1,8 +1,89 @@
-.. include:: /../common/stub-topic.txt
+Model Binding
+=============
 
-|stub-icon| Model Binding Request Data
-======================================
+By `Rachel Appel <http://github.com/rachelappel>`_
 
-.. include:: /../common/stub-notice.txt
+In this article
 
-.. _issue: https://github.com/aspnet/Docs/issues/114
+- `Introduction to model binding`_
+- `How model binding works`_
+- `Customize model binding behavior with attributes`_
+- `Binding formatted data from the request body`_ 
+
+Introduction to model binding
+-----------------------------
+Model binding in MVC maps data from HTTP requests to action method parameters. The parameters may be simple types such as strings, integers, or floats, or they may be complex classes. This is a great feature of MVC because mapping incoming data to a counterpart is an often repeated scenario, regardless of size or complexity of the data. MVC solves this problem by abstracting binding away so developers don't have to keep rewriting a slightly different version of that same code in every app. Writing your own text to type converter code is tedious, and error prone. 
+
+How model binding works
+-----------------------
+When MVC receives an HTTP request, it routes it to a specific action method of a controller. It determines which action method to run based on what is in the route data, then it binds values from the HTTP request to that action method's parameters. For example, consider the following URL:
+
+`http://contoso.com/movies/edit/2`
+
+Since the routing template looks like this, ``{controller=Home}/{action=Index}/{id?}``, ``movies/edit/2`` routes to the ``Movies`` controller, and its ``Edit`` action method. It also accepts an optional parameter called ``id``. This is due to MVC's use of conventions. So the code for the action method should look something like this: 
+
+.. literalinclude:: model-binding/sample/src/MVCMovie/Controllers/MoviesController.cs
+   :language: c#
+   :lines: 61-76  
+   
+.. note:: The strings in the URL route are not case sensitive.
+
+MVC checks for model binding attributes applied to the action method's parameters. If it finds a parameter attribute it will use the data source specified by the attribute. Otherwise, for any property that has a string converter, it looks for a value with a matching name. In the above example the only matching item is an integer named `id`, so then the value provider associates its values with action method parameters. Below is a list of the data sources in the order that model binding looks through them:
```

Comment:
I still think this paragraph dives too deep into how model binding is implemented. Instead I think we should introduce the high level ideas.I would replace this text with something like this:

MVC will try to bind request data to the action parameters by name. MVC will look for values for each parameter using the parameter name and the names of its public settable properties. In the above example, the only action parameter is named ``id``, which MVC binds to the value with the same name in the route values. In addition to route values MVC will bind data from various parts of the request and it does so in a set order. Below is a list of the data sources in the order that model binding looks through them: